File: Dddml.Wms.JavaCommon/src/org/dddml/wms/domain/OrganizationStructureTypeStateEventIdFlattenedDto.java
package org.dddml.wms.domain;

import java.util.HashMap;
import java.util.Map;

public class OrganizationStructureTypeStateEventIdFlattenedDto
{


    public static final String[] propertyNames = new String[]{
            "id",
            "version",
        };

    public static final String[] propertyTypes = new String[]{
            "String",
            "Long",
        };

    public static final Map<String, String> propertyTypeMap;

    static {
        propertyTypeMap = new HashMap<String, String>();
        for (int i = 0; i < propertyNames.length; i++ ) {
            propertyTypeMap.put(propertyNames[i], propertyTypes[i]);
        }
    }
	
    private OrganizationStructureTypeStateEventId value;

    public OrganizationStructureTypeStateEventIdFlattenedDto()
    {
        this(new OrganizationStructureTypeStateEventId());
    }

    public OrganizationStructureTypeStateEventIdFlattenedDto(OrganizationStructureTypeStateEventId value)
    {
        this.value = value;
    }

    public OrganizationStructureTypeStateEventId toOrganizationStructureTypeStateEventId()
    {
        return this.value;
    }


    public String getId()
    {
        return this.value.getId();
    }

    public void setId(String id)
    {
        this.value.setId(id);
    }

    public Long getVersion()
    {
        return this.value.getVersion();
    }

    public void setVersion(Long version)
    {
        this.value.setVersion(version);
    }


    @Override
    public boolean equals(Object obj)
    {
        if (obj == this) {
            return true;
        }
        if (obj == null || obj.getClass() != this.getClass()) {
            return false;
        }

        OrganizationStructureTypeStateEventIdFlattenedDto other = (OrganizationStructureTypeStateEventIdFlattenedDto)obj;
        return value.equals(other.value);
    }

    @Override
    public int hashCode()
    {
        return value.hashCode();
    }

}

File: Dddml.Wms.HttpServices.PhpClient/Dddml/Wms/Domain/OrganizationStructureTypeStateEventIdFlattenedDto.php
<?php

namespace Dddml\Wms\Domain;

use JMS\Serializer\Annotation\Type;
use Dddml\Serializer\Type\Long;
use Dddml\FlattenedDtoInterface;

class OrganizationStructureTypeStateEventIdFlattenedDto implements FlattenedDtoInterface
{

    const PROPERTIES = [
            'id' => 'string',
            'version' => 'Long',
        ];

    public static function getPropertyNames()
    {
        return array_keys(static::PROPERTIES);
    }

    public static function getPropertyTypes()
    {
        return array_values(static::PROPERTIES);
    }

	
    /**
     * @var OrganizationStructureTypeStateEventId
     */
    private $value;

    /**
     * @param OrganizationStructureTypeStateEventId $value
     */
    public function __construct(OrganizationStructureTypeStateEventId $value = null)
    {
        if ($value) {
            $this->value = $value;
        } else {
            $this->value = new OrganizationStructureTypeStateEventId();
        }
    }

    /**
     * @return OrganizationStructureTypeStateEventId
     */
    public function toOrganizationStructureTypeStateEventId()
    {
        return $this->value;
    }

    /**
     * @return string
     */
    public function getId()
    {
        return $this->value->getId();
    }

    /**
     * @param string $id
     */
    public function setId($id)
    {
        $this->value->setId($id);
    }

    /**
     * @return Long
     */
    public function getVersion()
    {
        return $this->value->getVersion();
    }

    /**
     * @param Long $version
     */
    public function setVersion($version)
    {
        $this->value->setVersion($version);
    }

    /**
     * @return string
     */
    public function toString()
    {
        $pValues = [
            $this->getId(),
            $this->getVersion(),
        ];
        return implode(',', $pValues);
    }

    /**
     * @param string $data
     *
     * @return OrganizationStructureTypeStateEventIdFlattenedDto
     */
    public function fromString($data)
    {
        $pValues = explode(',', $data);
        $this->setId($pValues[0]);
        $this->setVersion($pValues[1]);
        return $this;
    }

}

